contracts/token/ERC20/extensions/ERC20Votes.sol

```diff
@@ -62,9 +63,22 @@ abstract contract ERC20Votes is IERC20Votes, ERC20Permit {
      */
     function getPriorVotes(address account, uint256 blockNumber) external view override returns (uint256) {
         require(blockNumber < block.number, "ERC20Votes::getPriorVotes: not yet determined");
+        return _checkpointsLookup(_checkpoints[account], blockNumber);
+    }
 
-        Checkpoint[] storage ckpts = _checkpoints[account];
+    /**
+     * @dev Determine the totalSupply at the begining of `blockNumber`. Note, this value is the sum of all balances.
+     * It is but NOT the sum of all the delegated votes!
+     */
+    function getPriorTotalSupply(uint256 blockNumber) external view override returns(uint256) {
+        require(blockNumber < block.number, "ERC20Votes::getPriorTotalSupply: not yet determined");
+        return _checkpointsLookup(_totalSupplyCheckpoints, blockNumber);
+    }
 
+    /**
+     * @dev Lookup a value in a list of (sorted) checkpoints.
+     */
+    function _checkpointsLookup(Checkpoint[] storage ckpts, uint256 blockNumber) internal view virtual returns (uint256) {
```

Comment:
I'm concerned about this being internal because it gives access to a storage pointer, which would allow writing and defeat the purpose of private storage variables.

```suggestion
    function _checkpointsLookup(Checkpoint[] storage ckpts, uint256 blockNumber) private view virtual returns (uint256) {
```